builder/dockerfile/dispatchers.go

```diff
@@ -350,8 +354,7 @@ func dispatchRun(d dispatchRequest, c *instructions.RunCommand) error {
 	runConfigForCacheProbe := copyRunConfig(stateRunConfig,
 		withCmd(saveCmd),
 		withEntrypointOverride(saveCmd, nil))
-	hit, err := d.builder.probeCache(d.state, runConfigForCacheProbe)
-	if err != nil || hit {
+	if hit, err := d.builder.probeCache(d.state, runConfigForCacheProbe); err != nil || hit {
```

Comment:
These refactors should be separate from the fix

Reply:
Ah, the fix is not needed any more, so we just refactor and cleanup, then apply other commands which don't need the temp container creation any more.